python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -298,6 +305,51 @@ def _(py_val: int, dtype: DataType | None):
         c_obj = make_numeric_scalar(c_dtype.c_obj)
         (<numeric_scalar[uint64_t]*>c_obj.get()).set_value(py_val)
 
+    elif tid == type_id.DURATION_NANOSECONDS:
```

Comment:
Why did we have to add all of these extra conversions to duration types in this function in this PR?

Reply:
Through running the polars unit tests with `cudf_polars`, I discovered more necessary conversions now that we're using `Scalar.from_py` now - namely the ones described in the top post https://github.com/rapidsai/cudf/pull/18563#issue-3015477402